@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -79,55 +79,216 @@ export interface VisibleTileSetOptions {
+        this.m_tileCache.canEvict = (_, tile) => {
             // Tiles can be evicted that weren't requested in the last frame.
             return !tile.isVisible;
         };
     }
 
+    /**
+     * Get information how cached tiles affects cache space available.
+     *
+     * The way how cache evaluates the __resources size__ have a big influence on entire
+     * caching mechanism, if [[resourceComputationType]] is set to:
+     * [[ResourceComputationType.EstimationInMb]] then each tiles contributes to cache size
+     * differently depending on the memory consumed, on other side
+     * [[ResourceComputationType.NumberOfTiles]] says each tile occupies single slot in cache,
+     * so its real memory consumed does not matter affect caching behavior. Of course in
+     * the second scenario cache may grow significantly in terms of memory usage and thus it
+     * is out of control.
+     *
+     * @return [[ResourceComputationType]] enum that describes if resources are counted by
+     * space occupied in memory or just by number of them.
+     */
+    get resourceComputationType(): ResourceComputationType {
+        return this.m_resourceComputationType;
+    }
+
+    /**
+     * Get the cache capacity measured as number if megabytes or number of entries.
+     *
+     * The total cached tiles size determines cache saturation, if it reaches the capacity value
+     * then the resources becomes evicted (released) starting from the oldest (the latest used).
+     *
+     * @see size.
+     * @see resourceComputationType.
+     */
+    get capacity(): number {
+        return this.m_tileCache.capacity;
+    }
+
+    /**
+     * Get total cache size described as number of megabytes consumed or number of tiles stored.
+     *
+     * @see capacity.
+     * @see resourceComputationType.
+     */
+    get size(): number {
+        return this.m_tileCache.size;
+    }
+
+    /**
+     * Set cache capacity and the algorithm used for cache size calculation.
+     *
+     * @see capacity.
+     * @see resourceComputationType.
+     * @param size The new capacity declared in megabytes or number of entires.
+     * @param rct The enum value that determines how size and capacity are evaluated.
+     */
+    setCapacity(size: number, rct: ResourceComputationType) {
+        this.m_resourceComputationType = rct;
+        this.m_tileCache.setCapacityAndMeasure(size, (tile: Tile) => {
+            if (this.m_resourceComputationType === ResourceComputationType.EstimationInMb) {
+                // Default is size in MB.
+                return tile.memoryUsage * MB_FACTOR;
+            } else {
+                return 1;
+            }
+        });
+    }
+
+    /**
+     * Get tile cached or __undefined__ if tile is not yet in cache.
+     *
+     * @param tileCode En unique tile code (its morton code).
+     * @param dataSource A [[DataSource]] the tile comes from.
+     */
+    get(tileCode: number, dataSource: DataSource): Tile | undefined {
```

Comment:
I would prefer if this method would explicitely get all ingredients to compute the cache key, i.e.:
`get(mortonCode: number, offset:number, dataSource: DataSource): Tile | undefined`
Since the cache key is a string we could also remove the hack that adds the offset to the mortonCode

Reply:
Hmm.. sounds good!

Reply:
Ok done in separate patchset: 
https://github.com/heremaps/harp.gl/pull/884